grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1056,12 +1056,14 @@ class StockMovementService {
                 it.delete()
             }
             picklist.save()
-        }
 
-        // Save requisition item before PA refresh
-        requisitionItem.save(flush: true)
+            // Save requisition item before PA refresh
+            requisitionItem.save(flush: true)
 
-        productAvailabilityService.refreshProductsAvailability(requisitionItem?.requisition?.origin?.id, [requisitionItem?.product?.id], false)
+            productAvailabilityService.refreshProductsAvailability(requisitionItem?.requisition?.origin?.id, [requisitionItem?.product?.id], false)
```

Comment:
Is it possible / practical to run this all at once at the end (like up after the lose brace on Line 1030). Would that help?

Reply:
@jmiranda I think that would be possible, however, it's a hotfix and because it's already tested by Katarzyna and UATed by Kelsey, I'd restrain from further refactoring, and I'd push refactor of this part to the 0.8.19.